crypto/poly1305/poly1305_ameth.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2007-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include "internal/cryptlib.h"
+#include <openssl/evp.h>
+#include "internal/asn1_int.h"
+#include "internal/poly1305.h"
+#include "poly1305_local.h"
+
+#define POLY1305_TEST_PRIVATE_KEY_FORMAT
+
+/*
+ * POLY1305 "ASN1" method. This is just here to indicate the maximum POLY1305 output
+ * length and to free up an POLY1305 key.
+ */
+
+static int poly1305_size(const EVP_PKEY *pkey)
+{
+    return POLY1305_DIGEST_SIZE;
+}
+
+static void poly1305_key_free(EVP_PKEY *pkey)
+{
+    ASN1_OCTET_STRING *os = EVP_PKEY_get0(pkey);
+    if (os) {
+        if (os->data != NULL)
+            OPENSSL_cleanse(os->data, os->length);
+        ASN1_OCTET_STRING_free(os);
+    }
+}
+
+static int poly1305_pkey_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
+{
+    /* ASN1_PKEY_CTRL_DEFAULT_MD_NID not supported */
```

Comment:
I think the importance is that in many of these EVP_PKEY-type of methods, is that there is an MD to be selected, and in this case, it does not apply. I'm going to leave a modified comment.